### ISSUE TITLE
DateTime from Long.MIN_VALUE or MAX_VALUE

### ISSUE THREAD

**Issue by mstevens83** [JodaOrg/joda-time#190]
In joda-time version 2.3 both this ...
```java
DateTime minDT = new DateTime(Long.MIN_VALUE);
```
... and this ...
```java
DateTime maxDT = new DateTime(Long.MAX_VALUE);
```
... worked perfectly fine.

But since version 2.4 (and still in 2.5) either expression causes an `ArithmeticException` to be thrown:
```java
Exception in thread "main" java.lang.ArithmeticException: Adding time zone offset caused overflow
	at org.joda.time.DateTimeZone.convertUTCToLocal(DateTimeZone.java:965)
	at org.joda.time.chrono.ZonedChronology$ZonedDateTimeField.get(ZonedChronology.java:422)
	at org.joda.time.base.BaseDateTime.<init>(BaseDateTime.java:129)
	at org.joda.time.base.BaseDateTime.<init>(BaseDateTime.java:97)
	at org.joda.time.DateTime.<init>(DateTime.java:209)
	at MinMaxTest.main(MinMaxTest.java:29)
```

**Comment by AmazingDonut** [comment]
I am facing this problem as well. It's hard to adopt this change from a backwards compatibility perspective, since I already have a number of dates and ranges created using Long.MAX_VALUE. 

Moreover, I discovered that only the DateTime constructor was affected by this change. For example, it is still possible to create an Interval with Long.MAX_VALUE, which leads to hard-to-detect, possible runtime errors.
```java
Interval bigInterval = new Interval(0,Long.MAX_VALUE);
System.out.println(bigInterval);
System.out.println(bigInterval.getEnd());
```
Output:
```java
1969-12-31T16:00:00.000-08:00/292278994-08-17T00:12:55.807-07:00
Exception thrown

org.joda.time.IllegalFieldValueException: Value 292278994 for year must be in the range [-292275054,292278993]
	at org.joda.time.field.FieldUtils.verifyValueBounds(FieldUtils.java:234)
	at org.joda.time.chrono.BasicYearDateTimeField.set(BasicYearDateTimeField.java:83)
	at org.joda.time.chrono.ZonedChronology$ZonedDateTimeField.set(ZonedChronology.java:482)
	at org.joda.time.base.BaseDateTime.<init>(BaseDateTime.java:129)
	at org.joda.time.DateTime.<init>(DateTime.java:236)
	at org.joda.time.base.AbstractInterval.getEnd(AbstractInterval.java:83)
	at org.joda.time.ReadableInterval$getEnd.call(Unknown Source)
```

**Comment by jodastephen** [comment]
See issue #100

**Comment by jodastephen** [comment]
* I've tried capping the input at a safe value, but that causes errors all over the place.
* I've tried extending the range of valid years, but that causes even more errors.
* It isn't possible to add a fixed maximum valid instant because it varies by timezone.
* I could remove the validation that was added, but that would open up a class of bug.

TLDR, no solution likely here

**Comment by mstevens83** [comment]
Thanks for looking into this some more Stephen.
But what about the approach used in JSR-310/Java SE 8?
Are you sure there's no pair of (possibly "tighter") instant bounds that would work regardless of the time zone?

**Comment by jodastephen** [comment]
Doing so may be possible but it would be a lot of work. It would also break compatibility with even more users. My opinion is that the only acceptable solution would be to make Long.MAX_VALUE work, however doing so looks like an equally futile task, and certainly not one I've got time for!

**Comment by AmazingDonut** [comment]
I understand your reasoning, Stephen. What's your opinion on the Interval constructor issue I reported above?

**Comment by jodastephen** [comment]
Perhaps Interval should be made to match. It is certainly logical. Although as shown here, each fix can cause problems elsewhere!

**Comment by mstevens83** [comment]
Do you have any new thoughts on this? This problem means I've been suck at v2.3 for a long time.

**Comment by jodastephen** [comment]
Try the [min/max branch](https://github.com/JodaOrg/joda-time/commits/topic/minmax). I won't merge the changes unless I hear they are suitable.

**Comment by mstevens83** [comment]
Hi Stephen,

Thanks for spending time one this!
These changes satisfy my needs, and I agree with you that it is a sensible compromise to default to UTC when handling `Long.MAX_VALUE` and `Long.MIN_VALUE` value, even when a non-UTC `DateTimeZone` was passed to the `DateTime` constructor.

I've been running a bunch of tests (I've made my own, not relying on the min/max tests I saw you added in 295f43e).
My tests covered:
* the case I originally here: https://github.com/JodaOrg/joda-time/issues/190#issue-45026385 (+ similar cases);
* the `Interval` case reported above by @AmazingDonut: https://github.com/JodaOrg/joda-time/issues/190#issuecomment-71255053 (+ similar cases);
* the earlier related issue reported by @martin-a: https://github.com/JodaOrg/joda-time/issues/100#issue-24336086 (+ `Long_MIN_VALUE` version of that call);
* another related issue reported by @martin-a: https://github.com/JodaOrg/joda-time/issues/297#issue-93443393 (+ `Long_MIN_VALUE` version of that call);
* A few other things I could think of.

All these tests passed. So if you ask me the min/max branch is ready to be merged.

By the way, when do you think we could expect a new Joda-time release with these changes? Oh and would that release also include tzdata 2015g which was released earlier this month?

Best & thanks again!

Matthias

**Comment by jodastephen** [comment]
Thanks for testing. When a new release occurs it will include the latest time-zone data.
